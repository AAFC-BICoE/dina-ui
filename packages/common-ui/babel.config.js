module.exports = function(api) {
  api.cache(true);

  const presets = [
    "@babel/preset-env",
    "@babel/preset-typescript",
    "@babel/preset-react"
  ];
  const plugins = [
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-transform-runtime",
    "react-require"
  ];

  return { presets, plugins };
};
