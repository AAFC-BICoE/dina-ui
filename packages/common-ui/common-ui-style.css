/*
* CSS classes for elements that common across multiple pages and/or styled in JS files (i.e. styled after the CSS file loaded)
*/

/*
 * Changes the default color of the badge to dark grey. 
 * Currently there is no background color that is applied.
 */
.badge {
  background-color: #333333;
}

/*
* Fixes WCAG contrast issue for react-select input elements
* Not sure if why this works...perhaps padding the element renders it sufficiently for a contrast calculation.
*/
input[id^="react-select"] {
  padding: 1px !important;
}

/*
* Fixes WCAG contrast issue for placeholder text in various textfields
*/
.css-1wa3eu0-placeholder {
  color: #333333 !important;
}

.fieldset-h2-adjustment {
  font-size: 1.7rem !important;
  margin-top: 15px !important;
  margin-bottom: 10px !important;
}

.modal-header {
  justify-content: center;
}

/* Give the "Multiple Values" placeholders an italic font. */
.has-multiple-values :placeholder-shown,
.has-multiple-values .placeholder-text,
.has-multiple-values .react-select__placeholder {
  color: rgb(87, 120, 94);
  font-style: italic;
}

.associated-sample-link {
  background-color: #e9ecef;
}

/* HIghlight the Edit All tab's bulk-edited fields in green. */
.has-bulk-edit-value label {
  color: rgb(0, 110, 0);
}
.has-bulk-edit-value input[type="text"],
.has-bulk-edit-value input[type="text"]:focus,
.has-bulk-edit-value input[type="search"],
.has-bulk-edit-value input[type="search"]:focus,
.has-bulk-edit-value textarea,
.has-bulk-edit-value textarea:focus,
.has-bulk-edit-value .associated-sample-link,
.has-bulk-edit-value .react-select__control {
  border-color: green;
  background-color: rgb(175, 255, 175);
}
