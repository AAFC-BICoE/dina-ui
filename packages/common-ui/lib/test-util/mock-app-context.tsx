import { mount } from "enzyme";
import { merge, noop } from "lodash";
import { useMemo, useRef } from "react";
import { SWRConfig } from "swr";
import { PartialDeep } from "type-fest";
import { AccountContextI, AccountProvider } from "../account/AccountProvider";
import {
  ApiClientI,
  ApiClientImpl,
  ApiClientProvider
} from "../api-client/ApiClientContext";
import { CommonUIIntlProvider } from "../intl/common-ui-intl";
import { ModalProvider } from "../modal/modal";

interface MockAppContextProviderProps {
  apiContext?: PartialDeep<ApiClientI>;
  accountContext?: Partial<AccountContextI>;
  children?: React.ReactNode;
}

/**
 * Wraps a test-rendered component to provide the contexts that would be available in
 * the application.
 */
export function MockAppContextProvider({
  accountContext,
  apiContext = { apiClient: { get: () => undefined as any } },
  children
}: MockAppContextProviderProps) {
  const DEFAULT_MOCK_ACCOUNT_CONTEXT: AccountContextI = useMemo(
    () => ({
      authenticated: true,
      groupNames: ["aafc", "cnc"],
      initialized: true,
      login: noop,
      logout: noop,
      roles: ["user"],
      token: "test-token",
      username: "test-user",
      isAdmin: false
    }),
    []
  );

  const DEFAULT_API_CONTEXT_VALUE = useMemo(
    () =>
      new ApiClientImpl({
        newId: () => "00000000-0000-0000-0000-000000000000"
      }),
    []
  );

  const apiContextWithWarnings = {
    ...apiContext,
    // Add a warning when bulkGet doesn't return anything in a test:
    bulkGet: async (paths: string[], options) => {
      const resources = await apiContext?.bulkGet?.(paths, options);
      if (!resources) {
        console.warn("No response returned for bulkGet paths: ", paths);
      } else {
        resources.forEach((resource, index) => {
          if (!resource) {
            console.warn("No value returned for bulkGet path: ", paths[index]);
          }
        });
      }
      return resources;
    }
  };

  const modalWrapperRef = useRef<HTMLDivElement>(null);

  return (
    <SWRConfig
      // Reset SWR cache between tests.
      value={{ provider: () => new Map() }}
    >
      <AccountProvider
        value={{ ...DEFAULT_MOCK_ACCOUNT_CONTEXT, ...accountContext }}
      >
        <ApiClientProvider
          value={merge({}, DEFAULT_API_CONTEXT_VALUE, apiContextWithWarnings)}
        >
          <CommonUIIntlProvider>
            <div ref={modalWrapperRef}>
              <ModalProvider appElement={modalWrapperRef.current}>
                {children}
              </ModalProvider>
            </div>
          </CommonUIIntlProvider>
        </ApiClientProvider>
      </AccountProvider>
    </SWRConfig>
  );
}

/**
 * Helper function to get a test wrapper with the required context providers.
 */
export function mountWithAppContext(
  element: React.ReactNode,
  mockAppContextProviderProps?: MockAppContextProviderProps
) {
  return mount(
    <MockAppContextProvider {...mockAppContextProviderProps}>
      {element}
    </MockAppContextProvider>
  );
}
