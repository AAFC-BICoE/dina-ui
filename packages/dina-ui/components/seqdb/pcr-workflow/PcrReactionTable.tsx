import {
  AutoSuggestTextField,
  FieldHeader,
  filterBy,
  useApiClient,
  useDinaFormContext
} from "common-ui";
import { PersistedResource } from "kitsu";
import { sortBy } from "lodash";
import { useEffect, useState } from "react";
import ReactTable, { Column } from "react-table";
import { MaterialSample } from "../../../types/collection-api";
import {
  PcrBatchItem,
  PcrBatchItemDropdownResults,
  pcrBatchItemResultColor
} from "../../../types/seqdb-api";
import { getScientificNames } from "../../collection/material-sample/organismUtils";

export function usePcrReactionData(pcrBatchId?: string) {
  const [pcrBatchItems, setPcrBatchItems] = useState<PcrBatchItem[]>([]);
  const [materialSamples, setMaterialSamples] = useState<MaterialSample[]>([]);
  const [loading, setLoading] = useState<boolean>(true);

  const { apiClient, bulkGet } = useApiClient();

  // Initial fetch, should refetch between edit modes.
  useEffect(() => {
    fetchPcrBatchItems();
  }, [pcrBatchId]);

  function fetchPcrBatchItems() {
    if (!pcrBatchId) {
      setLoading(false);
    } else {
      setLoading(true);
      apiClient
        .get<PcrBatchItem[]>("seqdb-api/pcr-batch-item", {
          filter: filterBy([], {
            extraFilters: [
              {
                selector: "pcrBatch.uuid",
                comparison: "==",
                arguments: pcrBatchId
              }
            ]
          })(""),
          include: "materialSample",
          page: {
            limit: 1000 // Maximum page limit
          }
        })
        .then((response) => {
          const batchItems: PersistedResource<PcrBatchItem>[] =
            response.data?.filter(
              (item) => item?.materialSample?.id !== undefined
            );
          setPcrBatchItems(batchItems);
          fetchMaterialSamples(batchItems);
        });
    }
  }

  function fetchMaterialSamples(batchItems: PersistedResource<PcrBatchItem>[]) {
    if (!batchItems || batchItems.length === 0) {
      setLoading(false);
      return;
    }
    bulkGet<MaterialSample>(
      batchItems.map(
        (item) =>
          "/material-sample/" + item?.materialSample?.id + "?include=organism"
      ),
      { apiBaseUrl: "/collection-api" }
    ).then((response) => {
      setMaterialSamples(response);
      setLoading(false);
    });
  }

  return { pcrBatchItems, setPcrBatchItems, materialSamples, loading };
}

export interface PcrReactionTableProps {
  pcrBatchItems: PcrBatchItem[];
  materialSamples: MaterialSample[];
}

export function PcrReactionTable({
  pcrBatchItems,
  materialSamples
}: PcrReactionTableProps) {
  const { readOnly } = useDinaFormContext();

  const PCR_REACTION_COLUMN: Column<PcrBatchItem>[] = [
    {
      Cell: ({ original }) => {
        if (original?.wellRow === null || original?.wellColumn === null)
          return <></>;

        return <>{original.wellRow + "" + original.wellColumn}</>;
      },
      Header: <FieldHeader name={"wellCoordinates"} />,
      sortable: false
    },
    {
      Cell: ({ original }) => {
        if (original?.cellNumber === undefined) return <></>;

        return <>{original.cellNumber}</>;
      },
      id: "tubeNumber",
      Header: <FieldHeader name={"tubeNumber"} />,
      sortable: false
    },
    {
      Cell: ({ original }) => {
        const fetchedMaterialSample = materialSamples.find(
          (materialSample) => materialSample.id === original?.materialSample?.id
        );

        if (!fetchedMaterialSample) return <></>;

        return <p>{fetchedMaterialSample.materialSampleName}</p>;
      },
      Header: <FieldHeader name={"materialSampleName"} />,
      sortable: false
    },
    {
      Cell: ({ original }) => {
        const fetchedMaterialSample = materialSamples.find(
          (materialSample) => materialSample.id === original?.materialSample?.id
        );

        if (!fetchedMaterialSample) return <></>;

        return <>{getScientificNames(fetchedMaterialSample)}</>;
      },
      Header: <FieldHeader name={"scientificName"} />,
      sortable: false
    },
    {
      Cell: ({ original }) => {
        return !readOnly ? (
          <div>
            <AutoSuggestTextField
              name={"results[" + original?.id + "]"}
              hideLabel={true}
              removeBottomMargin={true}
              customOptions={(searchTerm, _) => {
                return Object.values(PcrBatchItemDropdownResults).filter(
                  (resultString) =>
                    searchTerm
                      ? resultString
                          .toLowerCase()
                          .startsWith(searchTerm.toLowerCase())
                      : true
                );
              }}
              blankSearchBackend={"json-api"}
            />
          </div>
        ) : (
          <div
            style={{
              backgroundColor: "#" + pcrBatchItemResultColor(original?.result),
              borderRadius: "5px",
              paddingLeft: "5px"
            }}
          >
            {original?.result ?? ""}
          </div>
        );
      },
      Header: <FieldHeader name={"result"} />,
      sortable: false
    }
  ];

  return (
    <ReactTable<PcrBatchItem>
      className="react-table-overflow"
      columns={PCR_REACTION_COLUMN}
      data={sortBy(pcrBatchItems, "cellNumber")}
      minRows={1}
      pageSize={1000} // Maximum that the API will return.
      showPagination={false}
      sortable={false}
    />
  );
}
