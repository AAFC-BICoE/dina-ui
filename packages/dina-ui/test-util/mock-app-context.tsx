import {
  AccountContextI,
  AccountProvider,
  ApiClientI,
  ApiClientImpl,
  ApiClientProvider,
  ModalProvider
} from "common-ui";
import { mount } from "enzyme";
import { merge, noop } from "lodash";
import { useMemo, useRef } from "react";
import { DndProvider } from "react-dnd-cjs";
import HTML5Backend from "react-dnd-html5-backend-cjs";
import { SWRConfig } from "swr";
import { PartialDeep } from "type-fest";
import { FileUploadProviderImpl } from "../components/object-store/file-upload/FileUploadProvider";
import { DinaIntlProvider } from "../intl/dina-ui-intl";

interface MockAppContextProviderProps {
  apiContext?: PartialDeep<ApiClientI>;
  accountContext?: Partial<AccountContextI>;
  children?: React.ReactNode;
}

/**
 * Wraps a test-rendered component to provide the contexts that would be available in
 * the application.
 */
export function MockAppContextProvider({
  accountContext,
  apiContext = { apiClient: { get: () => undefined as any } },
  children
}: MockAppContextProviderProps) {
  const DEFAULT_MOCK_ACCOUNT_CONTEXT: AccountContextI = useMemo(
    () => ({
      authenticated: true,
      groupNames: ["aafc", "cnc"],
      initialized: true,
      login: noop,
      logout: noop,
      roles: ["user"],
      token: "test-token",
      username: "test-user",
      isAdmin: false
    }),
    []
  );

  const DEFAULT_API_CONTEXT_VALUE = useMemo(
    () =>
      new ApiClientImpl({
        newId: () => "00000000-0000-0000-0000-000000000000"
      }),
    []
  );

  const apiContextWithWarnings = {
    ...apiContext,
    // Add a warning when bulkGet doesn't return anything in a test:
    bulkGet: async (paths: string[], options) => {
      const resources = await apiContext?.bulkGet?.(paths, options);
      if (!resources) {
        console.warn("No response returned for bulkGet paths: ", paths);
      } else {
        resources.forEach((resource, index) => {
          if (!resource) {
            console.warn("No value returned for bulkGet path: ", paths[index]);
          }
        });
      }
      return resources;
    }
  };

  const modalWrapperRef = useRef<HTMLDivElement>(null);

  return (
    <SWRConfig
      // Reset SWR cache between tests.
      value={{ provider: () => new Map() }}
    >
      <AccountProvider
        value={{ ...DEFAULT_MOCK_ACCOUNT_CONTEXT, ...accountContext }}
      >
        <ApiClientProvider
          value={merge({}, DEFAULT_API_CONTEXT_VALUE, apiContextWithWarnings)}
        >
          <FileUploadProviderImpl>
            <DinaIntlProvider>
              <DndProvider backend={HTML5Backend}>
                <div ref={modalWrapperRef}>
                  <ModalProvider appElement={modalWrapperRef.current}>
                    {children}
                  </ModalProvider>
                </div>
              </DndProvider>
            </DinaIntlProvider>
          </FileUploadProviderImpl>
        </ApiClientProvider>
      </AccountProvider>
    </SWRConfig>
  );
}

/**
 * Helper function to get a test wrapper with the required context providers.
 */
export function mountWithAppContext(
  element: React.ReactNode,
  mockAppContextProviderProps?: MockAppContextProviderProps
) {
  return mount(
    <MockAppContextProvider {...mockAppContextProviderProps}>
      {element}
    </MockAppContextProvider>
  );
}
