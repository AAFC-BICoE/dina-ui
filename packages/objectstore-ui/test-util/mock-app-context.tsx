import {
  AccountContextI,
  AccountProvider,
  ApiClientContextI,
  AuthenticatedApiClientProvider,
  createContextValue,
  ModalProvider
} from "common-ui";
import { mount } from "enzyme";
import { merge, noop } from "lodash";
import { ObjectStoreIntlProvider } from "../intl/objectstore-intl";

interface MockAppContextProviderProps {
  apiContext?: Partial<ApiClientContextI>;
  accountContext?: Partial<AccountContextI>;
  children?: React.ReactNode;
}

/**
 * Wraps a test-rendered component to provide the contexts that would be available in
 * the application.
 */
export function MockAppContextProvider({
  accountContext,
  apiContext,
  children
}: MockAppContextProviderProps) {
  return (
    <AccountProvider
      value={{ ...DEFAULT_MOCK_ACCOUNT_CONTEXT, ...accountContext }}
    >
      <AuthenticatedApiClientProvider
        apiContext={merge({}, DEFAULT_API_CONTEXT_VALUE, apiContext)}
      >
        <ObjectStoreIntlProvider>
          <ModalProvider appElement={document.querySelector("body")}>
            {children}
          </ModalProvider>
        </ObjectStoreIntlProvider>
      </AuthenticatedApiClientProvider>
    </AccountProvider>
  );
}

/**
 * Helper function to get a test wrapper with the required context providers.
 */
export function mountWithAppContext(
  element: React.ReactNode,
  mockAppContextProviderProps?: MockAppContextProviderProps
) {
  return mount(
    <MockAppContextProvider {...mockAppContextProviderProps}>
      {element}
    </MockAppContextProvider>
  );
}

const DEFAULT_MOCK_ACCOUNT_CONTEXT: AccountContextI = {
  authenticated: true,
  groups: ["/aafc", "cnc"],
  initialized: true,
  login: noop,
  logout: noop,
  token: "test-token",
  username: "test-user"
};

const DEFAULT_API_CONTEXT_VALUE = createContextValue({
  getTempIdGenerator: () => () => "00000000-0000-0000-0000-000000000000"
});
