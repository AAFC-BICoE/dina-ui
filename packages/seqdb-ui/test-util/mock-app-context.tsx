import {
  ApiClientContext,
  ApiClientContextI,
  createContextValue
} from "common-ui";
import { mount } from "enzyme";
import { SeqdbIntlProvider } from "../intl/seqdb-intl";

interface MockAppContextProviderProps {
  apiContext?: ApiClientContextI;
  children?: React.ReactNode;
}

/**
 * Wraps a test-rendered component to provide the contexts that would be available in
 * the application.
 */
export function MockAppContextProvider({
  apiContext,
  children
}: MockAppContextProviderProps) {
  return (
    <ApiClientContext.Provider value={apiContext || createContextValue()}>
      <SeqdbIntlProvider>{children}</SeqdbIntlProvider>
    </ApiClientContext.Provider>
  );
}

/**
 * Helper function to get a test wrapper with the required context providers.
 */
export function mountWithAppContext(
  element: React.ReactNode,
  mockAppContextProviderProps?: MockAppContextProviderProps
) {
  return mount(
    <MockAppContextProvider {...mockAppContextProviderProps}>
      {element}
    </MockAppContextProvider>
  );
}
