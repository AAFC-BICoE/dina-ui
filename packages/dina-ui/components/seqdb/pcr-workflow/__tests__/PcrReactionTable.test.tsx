import { DinaForm } from "common-ui";
import { mountWithAppContext } from "../../../../test-util/mock-app-context";
import { PcrReactionTable } from "../PcrReactionTable";
import {
  MATERIAL_SAMPLES,
  PCR_BATCH_ITEMS
} from "../__mocks__/PcrWorkflowMocks";

describe("PcrReactionTable component", () => {
  beforeEach(jest.clearAllMocks);

  test("Table renders correctly with mocked API calls", async () => {
    const wrapper = mountWithAppContext(
      <DinaForm initialValues={{}} readOnly={true}>
        <PcrReactionTable
          materialSamples={MATERIAL_SAMPLES}
          pcrBatchItems={PCR_BATCH_ITEMS}
        />
      </DinaForm>
    );

    await new Promise(setImmediate);
    wrapper.update();

    // Ensure well coordinates are displayed correctly
    let rowNumber = 0;
    wrapper
      .find("ReactTable")
      .find(".rt-tr-group")
      .forEach((row) => {
        // Well coordinates
        expect(row.find(".rt-td").at(0).text()).toEqual(
          PCR_BATCH_ITEMS?.[rowNumber]?.wellRow +
            "" +
            PCR_BATCH_ITEMS?.[rowNumber]?.wellColumn
        );

        // Tube Number
        expect(row.find(".rt-td").at(1).text()).toEqual(
          "" + PCR_BATCH_ITEMS?.[rowNumber]?.cellNumber
        );

        // Primary ID
        expect(row.find(".rt-td").at(2).text()).toEqual(
          "" + MATERIAL_SAMPLES?.[rowNumber]?.materialSampleName
        );

        // Scientific Name
        expect(row.find(".rt-td").at(3).text()).toEqual("scientificName");

        // Result
        expect(row.find(".rt-td").at(4).html()).toMatchSnapshot(
          `result band for row number: ${rowNumber}`
        );
        rowNumber++;
      });
  });
});
