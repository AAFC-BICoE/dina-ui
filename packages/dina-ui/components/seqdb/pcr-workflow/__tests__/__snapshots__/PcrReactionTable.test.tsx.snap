// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 0 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(222, 252, 222); border-radius: 5px; padding-left: 5px;\\">Good Band</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 1 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(222, 252, 222); border-radius: 5px; padding-left: 5px;\\">Good Band</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 2 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(222, 252, 222); border-radius: 5px; padding-left: 5px;\\">Good Band</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 3 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(222, 252, 222); border-radius: 5px; padding-left: 5px;\\">Good Band</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 4 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(222, 252, 222); border-radius: 5px; padding-left: 5px;\\">Good Band</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 5 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(255, 250, 205); border-radius: 5px; padding-left: 5px;\\">Weak Band</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 6 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(255, 250, 205); border-radius: 5px; padding-left: 5px;\\">Weak Band</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 7 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(255, 250, 205); border-radius: 5px; padding-left: 5px;\\">Weak Band</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 8 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(255, 250, 205); border-radius: 5px; padding-left: 5px;\\">Weak Band</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 9 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(255, 250, 205); border-radius: 5px; padding-left: 5px;\\">Weak Band</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 10 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(234, 206, 222); border-radius: 5px; padding-left: 5px;\\">Multiple Bands</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 11 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(234, 206, 222); border-radius: 5px; padding-left: 5px;\\">Multiple Bands</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 12 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(234, 206, 222); border-radius: 5px; padding-left: 5px;\\">Multiple Bands</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 13 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(234, 206, 222); border-radius: 5px; padding-left: 5px;\\">Multiple Bands</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 14 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(234, 206, 222); border-radius: 5px; padding-left: 5px;\\">Multiple Bands</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 15 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(255, 192, 203); border-radius: 5px; padding-left: 5px;\\">Contaminated</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 16 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(255, 192, 203); border-radius: 5px; padding-left: 5px;\\">Contaminated</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 17 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(255, 192, 203); border-radius: 5px; padding-left: 5px;\\">Contaminated</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 18 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(255, 192, 203); border-radius: 5px; padding-left: 5px;\\">Contaminated</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 19 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(255, 192, 203); border-radius: 5px; padding-left: 5px;\\">Contaminated</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 20 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(220, 220, 220); border-radius: 5px; padding-left: 5px;\\">Smear</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 21 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(220, 220, 220); border-radius: 5px; padding-left: 5px;\\">Smear</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 22 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(220, 220, 220); border-radius: 5px; padding-left: 5px;\\">Smear</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 23 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(220, 220, 220); border-radius: 5px; padding-left: 5px;\\">Smear</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 24 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(220, 220, 220); border-radius: 5px; padding-left: 5px;\\">Smear</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 25 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(146, 168, 209); border-radius: 5px; padding-left: 5px;\\">Custom Result</div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 26 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(146, 168, 209); border-radius: 5px; padding-left: 5px;\\"></div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 27 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(146, 168, 209); border-radius: 5px; padding-left: 5px;\\"></div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 28 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(146, 168, 209); border-radius: 5px; padding-left: 5px;\\"></div></div>"`;

exports[`PcrReactionTable component Table renders correctly with mocked API calls: result band for row number: 29 1`] = `"<div class=\\"rt-td\\" role=\\"gridcell\\" style=\\"flex: 100 0 auto; width: 100px;\\"><div style=\\"background-color: rgb(146, 168, 209); border-radius: 5px; padding-left: 5px;\\"></div></div>"`;
